src/browser/modules/Frame/FrameTitlebar.tsx

```diff
@@ -256,16 +258,61 @@ function FrameTitlebar(props: FrameTitleBarProps) {
     props.reRun(frame, cmd)
   }
 
+  function onPreviewClick(e: React.MouseEvent) {
+    if (e.ctrlKey || e.metaKey) {
+      props.onTitlebarCmdClick(editorValue)
+    } else {
+      setRenderEditor(true)
+    }
+  }
+
+  const titleBarRef = useRef<HTMLDivElement>(null)
+  useEffect(() => {
+    // We want clicks outside the frame itself, not just the titlebar.
+    // Because of how the component tree is built (we don't have a
+    // reference to the full frame body) we'd need to pass
+    // a ref from each parent to avoid this dom traversal
+    function handleClickOutside(event: MouseEvent) {
+      if (!(event.target instanceof Element)) {
+        return
+      }
+      const insideFrame = titleBarRef.current
+        ?.closest('article')
+        ?.contains(event.target)
+
+      const insideMainWrapper = document
+        .getElementById(MAIN_WRAPPER_DOM_ID)
+        ?.contains(event.target)
+
+      if (!insideFrame && insideMainWrapper) {
+        // a really quick click will lose some of the last edits
+        const editorRefVal = editorRef.current?.getValue()
+        if (editorRefVal && editorRefVal !== editorValue) {
+          setEditorValue(editorRefVal)
+        }
+        setRenderEditor(false)
+      }
+    }
+
+    document.addEventListener('mouseup', handleClickOutside)
```

Comment:
nice, the change to mouseup instead of mousedown seems to fix the bug clicking another frame when a multi-statement frame was previously being edited! :-)